### PR TITLE
bug:change isDragReject true when maxFiles rejectDrag

### PR DESCRIPTION
**Pull request by Morteza-Jenabzadeh**
**What kind of change does this PR introduce?**

- [x] bugfix
- [ ] feature
- [ ] refactoring / style
- [ ] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [x] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [ ] Yes, I've updated the documentation
- [x] Not relevant

**Summary**
There is a bug in maxFiles feature that `isDragReject` doesn't change when DragEvent contains more than the max number of files
This bug reported #1017 

**Does this PR introduce a breaking change?**

shouldn't
**Other information**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -751,7 +751,7 @@ export function useDropzone(options = {}) {
   )
 
   const fileCount = draggedFiles.length
-  const isDragAccept = fileCount > 0 && allFilesAccepted({ files: draggedFiles, accept, minSize, maxSize, multiple })
+  const isDragAccept = fileCount > 0 && allFilesAccepted({ files: draggedFiles, accept, minSize, maxSize, multiple,maxFiles })
```
Comment (rxmarbles) [inline src/index.js:754]:
```suggestion
  const isDragAccept = fileCount > 0 && allFilesAccepted({ files: draggedFiles, accept, minSize, maxSize, multiple, maxFiles })
```

File: src/utils/index.js [hunk 2]
```diff
@@ -59,8 +59,8 @@ function isDefined(value) {
   return value !== undefined && value !== null
 }
 
-export function allFilesAccepted({ files, accept, minSize, maxSize, multiple }) {
-  if (!multiple && files.length > 1) {
+export function allFilesAccepted({ files, accept, minSize, maxSize, multiple,maxFiles }) {
+  if ((!multiple && files.length > 1) ||(multiple && maxFiles >= 1 &&  files.length > maxFiles) ) {
```
Comment (rxmarbles) [inline src/utils/index.js:63]:
```suggestion
export function allFilesAccepted({ files, accept, minSize, maxSize, multiple, maxFiles }) {
  if ((!multiple && files.length > 1) || (multiple && maxFiles >= 1 &&  files.length > maxFiles) ) {
```

**Review by rxmarbles — CHANGES_REQUESTED** [review]
LGTM just need to fix a few spacing issues

File: src/utils/index.js (hunk 2 above)
Comment (Morteza-Jenabzadeh) [reply src/utils/index.js:63]: I couldn't find a way to run linter to format staged files (husky should handle this, but it didn't), and there is no prettier config file
please give me a hint about it.

**Comment by rolandjitsu** [comment]
I think we're missing some tests, @Morteza-Jenabzadeh could you add some test cases for these changes?

**Comment by Morteza-Jenabzadeh** [comment]
> I think we're missing some tests, @Morteza-Jenabzadeh could you add some test cases for these changes?

I have added a test to check isDragReject.

File: src/utils/index.js [hunk 3]
```diff
@@ -59,8 +59,8 @@ function isDefined(value) {
   return value !== undefined && value !== null
 }
 
-export function allFilesAccepted({ files, accept, minSize, maxSize, multiple }) {
-  if (!multiple && files.length > 1) {
+export function allFilesAccepted({ files, accept, minSize, maxSize, multiple, maxFiles }) {
```
Comment (rolandjitsu) [inline src/utils/index.js:62]: Let's add some tests for this fn as well.

**Review by rolandjitsu — CHANGES_REQUESTED** [review]

File: src/utils/index.js (hunk 3 above)
Comment (Morteza-Jenabzadeh) [reply src/utils/index.js:62]: The test has been added.

**Review by rolandjitsu — APPROVED** [review]

File: src/utils/index.spec.js [hunk 4]
```diff
@@ -229,7 +229,26 @@ it('rejects file when single accept criteria as array', () => {
 })
 
 })
-
+describe('allFilesAccepted()', () => {
+
```
Comment (rolandjitsu) [inline src/utils/index.spec.js:233]: Remove this empty line.

File: src/utils/index.spec.js [hunk 5]
```diff
@@ -229,7 +229,26 @@ it('rejects file when single accept criteria as array', () => {
...
+    expect(utils.allFilesAccepted({ files, multiple: true})).toEqual(true)
+    expect(utils.allFilesAccepted({ files, multiple: true, maxFiles: 10 })).toEqual(true)
+    expect(utils.allFilesAccepted({ files, multiple: false, maxFiles: 10 })).toEqual(false)
+    expect(utils.allFilesAccepted({ files, multiple: true, accept:'image/jpeg' })).toEqual(false) 
+    expect(utils.allFilesAccepted({ files: images, multiple: true,accept:'image/*' })).toEqual(true) 
+    expect(utils.allFilesAccepted({ files, multiple: true, minSize: 110 })).toEqual(false)
+    expect(utils.allFilesAccepted({ files, multiple: true, maxSize: 99 })).toEqual(false)
+    expect(utils.allFilesAccepted({ files, multiple: true, maxFiles: 1 })).toEqual(false)
+  })
+})
```
Comment (rolandjitsu) [inline src/utils/index.spec.js:251]: Add a new empty line after this test.

File: src/index.spec.js [hunk 6]
```diff
@@ -2095,11 +2095,13 @@ describe('useDropzone() hook', () => {
     it('rejects all files if {multiple} is true and maxFiles is less than files and {accept} criteria is met', async () => {
       const onDropSpy = jest.fn()
       const onDropRejectedSpy = jest.fn()
-      const ui = (
-        <Dropzone accept="image/*" onDrop={onDropSpy} onDropRejected={onDropRejectedSpy} multiple={true} maxFiles={1}>
-          {({ getRootProps, getInputProps }) => (
+      const ui = ( 
+         <Dropzone accept="image/*" onDrop={onDropSpy} onDropRejected={onDropRejectedSpy} multiple={true} maxFiles={1}>
```
Comment (rolandjitsu) [inline src/index.spec.js:2099]: The whitespace here is off:

```suggestion
      const ui = (
        <Dropzone accept="image/*" onDrop={onDropSpy} onDropRejected={onDropRejectedSpy} multiple={true} maxFiles={1}>
```

**Review by rolandjitsu — CHANGES_REQUESTED** [review]
@Morteza-Jenabzadeh just a few more style changes and we're good to merge.

**Review by rolandjitsu — APPROVED** [review]

**Pull request merged**